src/mlpack/methods/ann/layer/adaptive_max_pooling_impl.hpp

```diff
@@ -33,6 +36,56 @@ AdaptiveMaxPooling<InputDataType, OutputDataType>::AdaptiveMaxPooling(
   // Nothing to do here.
 }
 
+template<typename InputDataType, typename OutputDataType>
+AdaptiveMaxPooling<InputDataType, OutputDataType>::AdaptiveMaxPooling(
+    const AdaptiveMaxPooling& layer) : 
+    poolingLayer(layer.poolingLayer),
+    outputWidth(layer.outputWidth),
+    outputHeight(layer.outputHeight),
+    reset(layer.reset)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+AdaptiveMaxPooling<InputDataType, OutputDataType>::AdaptiveMaxPooling(
+    AdaptiveMaxPooling&& layer) : 
+    poolingLayer(std::move(layer.poolingLayer)),
+    outputWidth(std::move(layer.outputWidth)),
+    outputHeight(std::move(layer.outputHeight)),
+    reset(std::move(layer.reset))
+{
+  // Nothing to do here.
+}
+
+template <typename InputDataType, typename OutputDataType>
+AdaptiveMaxPooling<InputDataType, OutputDataType>& 
+AdaptiveMaxPooling<InputDataType, OutputDataType> :: operator=(const AdaptiveMaxPooling& layer)
```

Comment:
```suggestion
AdaptiveMaxPooling<InputDataType, OutputDataType>::operator=(
  const AdaptiveMaxPooling& layer)
```

Remove some extra spaces and align with the 80 character line limit.